### PR TITLE
Allow accessing user defined value type members via contract name.

### LINKED ISSUE #11955: User defined type constructors not visible across non-inherited contracts
The following program compiles,

```
contract C { type T0 is uint; }
contract D { C.T0 x; }
```

but the next one does not.

```
contract C { type T0 is uint; }
contract D { C.T0 x = C.T0.wrap(uint(1)); }
```

Found by fuzzing.

### PR DESCRIPTION
**Pull request by ekpyron**
Seems to be as simple as this, ~~but I'll still add a few semantics tests just to be sure.~~ (added one)
Fixes https://github.com/ethereum/solidity/issues/11955.

### CODE REVIEW

**Review by hrkrshnn — DISMISSED** [review]

File: test/libsolidity/syntaxTests/userDefinedValueType/wrap_unwrap_via_contract_name_different.sol [hunk 1]
```diff
@@ -0,0 +1,8 @@
+contract C { type T is uint; }
+library L { type T is uint; }
+contract D
+{
+    C.T x = L.T.wrap(uint(1));
+}
+// ----
+// TypeError 7407: (86-103): Type user defined type T is not implicitly convertible to expected type user defined type T.
```
Comment (ekpyron) [inline test/libsolidity/syntaxTests/userDefinedValueType/wrap_unwrap_via_contract_name_different.sol:8]: Yeah, ok, this works, but the error message sucks :-D.

Comment (hrkrshnn) [reply test/libsolidity/syntaxTests/userDefinedValueType/wrap_unwrap_via_contract_name_different.sol:8]: How would you change it?

Comment (hrkrshnn) [reply test/libsolidity/syntaxTests/userDefinedValueType/wrap_unwrap_via_contract_name_different.sol:8]: Maybe should also include `L.T` and `C.T` in the `Type::toString()`.

**Review by chriseth — APPROVED** [review]

File: test/libsolidity/syntaxTests/userDefinedValueType/wrap_unwrap_via_contract_name_different.sol (hunk 1 above)
Comment (ekpyron) [reply test/libsolidity/syntaxTests/userDefinedValueType/wrap_unwrap_via_contract_name_different.sol:8]: If I were to change it, I'd do the same as we currently do for enums... but @chriseth and I just said, we don't really care, people should do this on file-level with nice names anyways :-).

Comment (hrkrshnn) [reply test/libsolidity/syntaxTests/userDefinedValueType/wrap_unwrap_via_contract_name_different.sol:8]: Here's the PR for this: https://github.com/ethereum/solidity/pull/11957

**Pull request merged**